src/main/java/io/github/mfoo/libyear/LibYearMojo.java

```diff
@@ -491,6 +511,70 @@ public void execute() throws MojoExecutionException {
         }
     }
 
+    private void generateReport(Set<Dependency> dependencies) {
+        if (StringUtils.isNotBlank(reportFile)) {
+
+            StringBuilder logsToReport = new StringBuilder();
+
+            if (!Paths.get(reportFile).toFile().exists()) {
+                logsToReport
+                        .append("All dependencies older than ")
+                        .append(minLibYearsForReport)
+                        .append(" libyears:")
+                        .append(System.lineSeparator())
+                        .append(System.lineSeparator());
+            }
+
+            dependencies.stream().forEach(dependency -> {
+                try {
+                    Artifact artifact = getHelper().createDependencyArtifact(dependency);
+
+                    Optional<LocalDate> currentReleaseDate = getReleaseDate(
+                            artifact.getGroupId(),
+                            artifact.getArtifactId(),
+                            artifact.getSelectedVersion().toString());
+
+                    String depName = dependency.getGroupId() + ":" + dependency.getArtifactId() + ": "
+                            + dependency.getVersion() + " ";
+                    if (!currentReleaseDate.isEmpty()) {
+                        long libWeeksOutdated = ChronoUnit.WEEKS.between(currentReleaseDate.get(), LocalDate.now());
+                        float libYearsOutdated = libWeeksOutdated / 52f;
+
+                        if (libYearsOutdated > 0
+                                && (minLibYearsForReport <= 0 || libYearsOutdated > minLibYearsForReport)) {
+                            String libYearsStr = String.format(" %.2f libyears", libYearsOutdated);
+                            addToReport(depName, libYearsStr, logsToReport);
+                        }
+                    } else {
+                        addToReport(depName, "unknown", logsToReport);
+                    }
+                } catch (MojoExecutionException | OverConstrainedVersionException e) {
+                    getLog().error("Exception by writing report", e);
+                }
+            });
+            Path path = Paths.get(reportFile);
+
+            try {
+                Files.write(
```

Comment:
If I run this on a multi-module project, with a path that would resolve the same for multiple projects (e.g. `/tmp/foo.txt` or `../foo.txt`, will the last one overwrite all of the previous ones?

Are there any concurrency issues if this was done in parallel?

Example parameters: `-DreportFile=/tmp/foo.txt -T2C`.

Would it be simpler to replace the text file writing here with log output, and to solve the file problem globally (separately)?

Reply:
In my opinion it shouldn't be a problem. The report is written in append mode.